samples/SimpleDataSource/HelloWorldCustomDataProcessor.cs

```diff
@@ -0,0 +1,159 @@
+            // For example, XPerf looks at what tables are enabled in order to turn on additional event processors to avoid processing everything if it doesn't have to.
+            //
+            // In this sample we create a Time Stamp column that can be used for graphing. For more information on graphing, go to "Graphing" on our Wiki. Link can be found in README.md
+            //
+
+            Timestamp startTime = Timestamp.MaxValue;
+            Timestamp endTime = Timestamp.MinValue;
+            DateTime firstEvent = DateTime.MinValue;
+            var contentDictionary = new Dictionary<string, IReadOnlyList<Tuple<Timestamp, string>>>();
+
+            foreach (var path in this.filePaths)
+            {
+                var list = new List<Tuple<Timestamp, string>>();
+                var content = System.IO.File.ReadAllLines(path);
+                
+
+                foreach (var line in content)
+                {
+                    var items = line.Split(new[] { ',' }, 2);
+
+                    if (items.Length < 2)
+                    {
+                        throw new InvalidOperationException("File line cannot be split to two sub-strings");
+                    }
+
+                    DateTime time;
+                    if (!DateTime.TryParse(items[0], out time))
+                    {
+                        throw new InvalidOperationException("Time cannot be pasred to DateTime format");
+                    }
+
+                    var timeStamp = Timestamp.FromNanoseconds(time.Ticks * 100);
+                    var words = items[1];
+
+                    if (timeStamp < startTime)
+                    {
+                        startTime = timeStamp;
+                        firstEvent = time;
+                    }
+
+                    if (timeStamp > endTime)
+                    {
+                        endTime = timeStamp;
+                    }
+
+                    list.Add(new Tuple<Timestamp, string>(timeStamp, words));
+                }
+
+                contentDictionary[path] = list.AsReadOnly();
+                this.dataSourceInfo = new DataSourceInfo(startTime.ToNanoseconds, endTime.ToNanoseconds, firstEvent.ToUniversalTime());
+            }
+
+            this.fileContent = new ReadOnlyDictionary<string, IReadOnlyList<Tuple<Timestamp, string>>>(contentDictionary);
+
+            return Task.CompletedTask;
+        }
+
+        protected override void BuildTableCore(
+            TableDescriptor tableDescriptor,
+            Action<ITableBuilder, IDataExtensionRetrieval> tableType,
+            ITableBuilder tableBuilder)
+        {
+            //
+            // Instantiate the table, and pass the tableBuilder to it.
+            //
+
+            var type = tableDescriptor.ExtendedData["Type"] as Type;
+
+            if (type != null)
```

Comment:
When building a table with this method, how does the SDK actually know if the table is instantiated?

Reply:
The ITableBuilder passed in has a method that is called when the table is built, which signals to the SDK that a table has been created